File: frontend/types/conversation.ts
import type { Message } from "./chat";

export interface Conversation {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface ConversationDetail extends Conversation {
  messages: Message[];
}

export interface ConversationTokenUsage {
  current_tokens: number;
  max_conversation_tokens: number;
  remaining_tokens: number;
  utilization_percent: number;
  chat_context_window_tokens: number;
  max_chat_input_tokens: number;
  max_chat_output_tokens: number;
  estimated: boolean;
}

File: backend/app/schemas/conversation.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    role: str
    content: str
    created_at: datetime


class ConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    created_at: datetime
    updated_at: datetime


class ConversationDetail(ConversationResponse):
    messages: list[MessageResponse]


class ConversationTokenUsage(BaseModel):
    current_tokens: int
    max_conversation_tokens: int
    remaining_tokens: int
    utilization_percent: float
    chat_context_window_tokens: int
    max_chat_input_tokens: int
    max_chat_output_tokens: int
    estimated: bool = True
